retro/reco.py

```diff
@@ -150,13 +181,49 @@ def run(self, method):
             t1 = time.time()
 
             # dump
-            llhp = np.empty(shape=len(param_values), dtype=llhp_t)
+            llhp = np.zeros(shape=len(param_values), dtype=llhp_t)
             llhp['llh'] = log_likelihoods
             llhp[dim_names] = param_values
 
-            llhp_outf = self.out_prefix + 'llhp.npy'
-            print('Saving llhp to "{}"...'.format(llhp_outf))
-            np.save(llhp_outf, llhp)
+            
+            # create derived dimensions
+            if 'energy' in derived_dim_names:
+                if 'track_energy' in dim_names:
+                    llhp['energy'] += llhp['track_energy']
+                if 'cascade_energy' in dim_names:
+                    llhp['energy'] += llhp['cascade_energy']
+
+            if 'cascade_d_zenith' in dim_names and 'cascade_d_azimuth' in dim_names:
+                # create cascade angles from delta angles
+                rotate_point(llhp['cascade_d_zenith'], llhp['cascade_d_azimuth'],
+                             llhp['track_zenith'], llhp['track_azimuth'],
+                             llhp['cascade_zenith'], llhp['cascade_azimuth'])
+
+            if 'track_zenith' in all_dim_names and 'track_azimuth' in all_dim_names:
+                if 'cascade_zenith' in all_dim_names and 'cascade_azimuth' in all_dim_names:
+                    # this resulting radius we won't need, but need to supply an array to the function
+                    r_out = np.empty(shape=llhp.shape, dtype=np.float32)
+                    # combine angles:
+                    add_vectors(llhp['track_energy'], llhp['track_zenith'], llhp['track_azimuth'],
+                                llhp['cascade_energy'], llhp['cascade_zenith'], llhp['cascade_azimuth'],
+                                r_out, llhp['zenith'], llhp['azimuth'])
+
+                else:
+                    # in this case there is no cascade angles
+                    llhp['zenith'] = llhp['track_zenith']
+                    llhp['azimuth'] = llhp['track_azimuth']
+
+            elif 'cascade_zenith' in all_dim_names and 'cascade_azimuth' in all_dim_names:
+                # in this case there are no track angles
+                llhp['zenith'] = llhp['cascade_zenith']
+                llhp['azimuth'] = llhp['cascade_azimuth']
+
+
+            if False:
```

Comment:
create a global var up top e.g. `SAVE_FULL_INFO = False` whereby this behavior can be changed but it's clear to someone reading the code where/how/what is intended to be modifiable behavior of the module without having to read all of the code of the module